vendor/models/invoice.py

```diff
@@ -267,6 +267,11 @@ def save_discounts_vendor_notes(self):
         self.vendor_notes["discounts"] = self.get_discounts()
         self.save()
 
+    def get_promos(self):
+        if 'promos' not in self.vendor_notes or not len(self.vendor_notes.get('promos', 0)):
```

Comment:
if i'm not mistaken, the second check actually checks if promos exists and returns promos if so and 0 if not; then checks the length of the result. getting len(0) will cause an error i think, but you shouldn't run into this case because of your first check. i would maybe remove the "0 as default" since it doesn't quite do what it's implying